analyser_merge_milestone_FR_metropole.py

```diff
@@ -0,0 +1,120 @@
+#                    filter = lambda text: text.replace(',', '.')), separator = u"\t"),
+# need to read this file in first : route + depPrD + prD is unique code match Gestionnaire => operator to second csv...?
+
+        self.init(
+            u"https://www.data.gouv.fr/fr/datasets/bornage-du-reseau-routier-national/",
+            u"Bornage du réseau routier national",
+            CSV(Source(u"data.gouv.fr:Ministère de la Transition écologique et solidaire", millesime = "01/2019",
+                    fileUrl = u"https://www.data.gouv.fr/fr/datasets/r/fbc8b73b-a65c-486b-a710-ed22b9e4070c",
+                    filter = lambda text: text.replace(',', '.')), separator = u"\t"),
+                    
+            Load("x", "y",srid = 2154,
+                where = lambda row: self.transform_to_natref(row) is not None),
+            Mapping(
+                select = Select(
+                    types = ["nodes"],
+                    tags = [{"highway": "milestone"}]),
+                osmRef = "nat_ref",    
+                conflationDistance = 150,
+                generate = Generate(
+                    static1 = {"highway": "milestone"},
+                    static2 = {"source": self.source},
+                    mapping1 = {
+                        "distance": 'pr',
+                        "nat_ref": lambda row: self.transform_to_natref(row),
+                        "ref": lambda fields: self.transform_route(fields['route']) }
+                    )))
+
+    def transform_to_natref(self,row):
+        #dept must be on 2 caracter
+        dept=row['depPr']
+        #C or '', not 'N'
+        if (row['concessionPr']=='C') :
+            concede = 'C'
+        else :
+            concede = ''
+        #I is for ignore, sens is D,G or U for droite(sens croissant),gauche(sens décroissant), unique. 
+        sens = row['cote']
+        if (row['cote']=='I') :
+            sens = 'U'
+            
+        if len(dept) == 1 :
+            dept = '0' + dept
+        if (dept in ('975','976','977')) :
+            #coordinate is in utm
+            return None
+        elif (dept==row['route'][0:2]) :
+            #not a milestone but highway junction
+            return None
+        elif (row['route'][0:1]=='P') :
+            #P for temporary,
+            return None
+        elif (row['route'][0:2] in ('N1','N2')) :
+            #N1 for future up_class and N2 for down_class road,
+            return None
+        else :
+            return dept + 'PR' + row['pr'] + sens + concede
+            
+    def transform_route(self,route):
+        #remove multiple 0 and add space
+        if   route[0:4] in ('A000', 'N000') : return route[0:1] + " " + route[4:]
+        elif route[0:3] in ('A00', 'N00')   : return route[0:1] + " " + route[3:]
+        elif route[0:2] in ('A0', 'N0')     : return route[0:1] + " " + route[2:]
+        else : return route[0:1] + " " + route[1:]
+
+
+class Sub_Analyser_Merge_Milestone_FR_gestionnaire(Analyser_Merge):
```

Comment:
This part is an other analyser? Sub_Analyser are for sub part of dynamic analyser. But I think it is not your case here.
We have new a doc, give a look at https://github.com/osm-fr/osmose-backend/tree/master/doc

Reply:
i don't Know how to do this :
  merge tag operator from one  CSV
to an other by an uniq calculated Key between the 2 csv

Reply:
It can be done. But first, let make it without this part.